### PR TITLE
Feat/connect programme listing mv to the 2023 curriculum api   LESQ-49

### PR DESCRIPTION
**Pull request by JBR90**
## Description

- Connects up programme listing to 2023 api

@jack-skerrett-oak - keyStageTitle was left out of the 2023 MV so i've added it into the acceptance criteria for the exam board programmes ticket.

Fixes #

## How to test

1. Go to ks3 science on 2023 to see example of tiers

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner

### CODE REVIEW

File: src/components/SubjectTierListing/SubjectTierListing.tsx [hunk 1]
```diff
@@ -3,35 +3,17 @@ import React, { FC } from "react";
...
+const SubjectTierListing: FC<ProgrammeListingPageData> = ({ ...props }) => {
+  const { programmes, keyStageSlug, subjectSlug, keyStageTitle = "" } = props;
+  const subjectTitle = programmes[0]?.subjectTitle ?? "";
 
   return (
     <>
       <TitleCard
         page={"subject"}
-        keyStage={keyStageTitle}
+        keyStage={keyStageTitle ?? ""}
```
Comment (JBR90) [inline src/components/SubjectTierListing/SubjectTierListing.tsx:16]: This was left out of the MV - i'm going to add it back in here:
https://www.notion.so/oaknationalacademy/I-want-to-connect-the-programme-listing-MV-to-the-2023-curriculum-API-ebc22f117c434ef4bb917ece325fe1c7

File: src/pages/beta/[viewType]/key-stages/[keyStageSlug]/subjects/[subjectSlug]/programmes.tsx [hunk 2]
```diff
@@ -16,27 +14,22 @@ import {
...
 
-export type ProgrammeListingPageProps = TierListingData;
+const ProgrammesListingPage: NextPage<ProgrammeListingPageData> = (props) => {
+  const {
+    programmes,
+    keyStageSlug,
+    subjectSlug,
+    keyStageTitle = "Key stage 3",
+  } = props;
+  const subjectTitle = programmes[0]?.subjectTitle ?? "";
```
Comment (JBR90) [inline src/pages/beta/[viewType]/key-stages/[keyStageSlug]/subjects/[subjectSlug]/programmes.tsx:27]: I'm going to add this as a field in the row on the MV so we destructure it with the other props. https://www.notion.so/oaknationalacademy/I-want-to-connect-the-programme-listing-MV-to-the-2023-curriculum-API-ebc22f117c434ef4bb917ece325fe1c7

File: src/pages/beta/[viewType]/key-stages/[keyStageSlug]/subjects/[subjectSlug]/programmes.tsx [hunk 3]
```diff
@@ -62,7 +55,7 @@ const ProgrammesListingPage: NextPage<ProgrammeListingPageProps> = (props) => {
                   viewType: "teachers",
                   keyStageSlug,
                 },
-                label: keyStageTitle,
+                label: keyStageTitle ?? "",
```
Comment (JBR90) [inline src/pages/beta/[viewType]/key-stages/[keyStageSlug]/subjects/[subjectSlug]/programmes.tsx:58]: Left out of MV - Will add this back in here - https://www.notion.so/oaknationalacademy/I-want-to-connect-the-programme-listing-MV-to-the-2023-curriculum-API-ebc22f117c434ef4bb917ece325fe1c7

**Review by benprotheroe — COMMENTED** [review]
any chance for a unit test for the query in here?

**Comment by JBR90** [comment]
> any chance for a unit test for the query in here?

Just added it :)

**Review by benprotheroe — APPROVED** [review]

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
lowercase "foundation"?

Screenshot 2023-06-30 at 09 18 50

**Comment by JBR90** [comment]
> lowercase "foundation"?
> 
> Screenshot 2023-06-30 at 09 18 50

Good spot, just checked the MV and it should be the 'description' field rather than 'tier' field - I've added to this ticket
https://www.notion.so/oaknationalacademy/I-want-exam-board-and-tier-on-the-mv_programme_listing-8a5ef942178342e8b60d814608670634

**Review by jack-skerrett-oak — APPROVED** [review]

**Pull request merged**